drivers/adgs1408/adgs1408.c

```diff
@@ -0,0 +1,474 @@
+				   uint8_t reg_addr,
+				   uint8_t mask,
+				   uint8_t *data)
+{
+	uint8_t reg_data;
+	int32_t ret;
+
+	if (dev->daisy_chain_en == ADGS1408_ENABLE) {
+		printf("%s: This feature is not available in Daisy-Chain mode.\n",
+		       __func__);
+		return FAILURE;
+	}
+
+	ret = adgs1408_spi_reg_read(dev, reg_addr, &reg_data);
+	*data = (reg_data & mask);
+
+	return ret;
+}
+
+/**
+ * SPI internal register write to device using a mask.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - The mask.
+ * @param data - The register data.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adgs1408_spi_reg_write_mask(struct adgs1408_dev *dev,
+				    uint8_t reg_addr,
+				    uint8_t mask,
+				    uint8_t data)
+{
+	uint8_t reg_data;
+	int32_t ret;
+
+	if (dev->daisy_chain_en == ADGS1408_ENABLE) {
+		printf("%s: This feature is not available in Daisy-Chain mode.\n",
+		       __func__);
+		return FAILURE;
+	}
+
+	ret = adgs1408_spi_reg_read(dev, reg_addr, &reg_data);
+	reg_data &= ~mask;
+	reg_data |= data;
+	ret |= adgs1408_spi_reg_write(dev, reg_addr, reg_data);
+
+	return ret;
+}
+
+/**
+ * Do a software reset.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adgs1408_do_soft_reset(struct adgs1408_dev *dev)
+{
+	int32_t ret;
+
+	if (dev->daisy_chain_en == ADGS1408_ENABLE) {
+		printf("%s: This feature is not available in Daisy-Chain mode.\n",
+		       __func__);
+		return FAILURE;
+	}
+
+	ret = adgs1408_spi_reg_write(dev,
+				     ADGS1408_REG_SOFT_RESETB,
+				     ADGS1408_RESET_1);
+	ret |= adgs1408_spi_reg_write(dev,
+				      ADGS1408_REG_SOFT_RESETB,
+				      ADGS1408_RESET_2);
+
+	return ret;
+}
+
+/**
+ * Clear the Error Flags Register.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adgs1408_clear_err_flags(struct adgs1408_dev *dev)
+{
+	uint8_t buf[3];
+	uint8_t buf_size = 2;
+
+	if (dev->daisy_chain_en == ADGS1408_ENABLE) {
+		printf("%s: This feature is not available in Daisy-Chain mode.\n",
+		       __func__);
+		return FAILURE;
+	}
+
+	buf[0] = ADGS1408_CLR_1;
+	buf[1] = ADGS1408_CLR_2;
+	if (dev->crc_en == ADGS1408_ENABLE) {
+		buf[2] = adgs1408_compute_crc8(&buf[0], 2);
+		buf_size = 3;
+	}
+	return spi_write_and_read(dev->spi_desc, buf, buf_size);
+}
+
+/**
+ * Enter Daisy-Chain Mode.
+ * @param dev - The device structure.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adgs1408_enter_daisy_chain(struct adgs1408_dev *dev)
+{
+	uint8_t buf[2];
+
+	if (dev->daisy_chain_en == ADGS1408_ENABLE) {
+		printf("%s: This feature is not available in Daisy-Chain mode.\n",
+		       __func__);
+		return FAILURE;
+	}
+
+	buf[0] = ADGS1408_DAISY_CHAIN_1;
+	buf[1] = ADGS1408_DAISY_CHAIN_2;
+	return spi_write_and_read(dev->spi_desc, buf, 2);
```

Comment:
the same comment as above; also, please check this in the rest of the code as well